### ISSUE TITLE
Add support for session_server

### ISSUE THREAD

**Issue by roock** [npalm/terraform-aws-gitlab-runner#92]
It would be cool if we could support session_server in the gitlab runner config for interactive webterminals

https://docs.gitlab.com/ee/ci/interactive_web_terminal/index.html
https://docs.gitlab.com/runner/configuration/advanced-configuration.html#the-session_server-section

**Comment by npalm** [comment]
@roock cool idea, any idea thoughts for an implementation?

**Comment by roock** [comment]
some thoughts:
* we would need to aquire an static Elastic IP address
* we could assign this IP with the following script: https://github.com/skymill/aws-ec2-assign-elastic-ip
* we would need an additiona parameter to add a rule to the security group for allowing Gitlab to the websession port

**Comment by kayman-mk** [comment]
Oh yes, a feature I missed already. Guess no work is going on at the moment, right?

**Comment by npalm** [comment]
Yep, no development is going on, feel free to create a PR

**Comment by kayman-mk** [comment]
The EIP is assigned to the runner agent. #124 will not be solved here.
I will also offer the possibility to add a rule to a load balancer instead of assigning an EIP.

**Comment by kayman-mk** [comment]
I reviewed my last comment and think that it makes no sense to add an IP or load balancer or something else. The reason is, that you should already been able to reach the runner from your gitlab instance (to start the build jobs).

**Comment by roock** [comment]
> I reviewed my last comment and think that it makes no sense to add an IP or load balancer or something else. The reason is, that you should already been able to reach the runner from your gitlab instance (to start the build jobs).

Afaik there is a difference between regular runner jobs and the session Server. For regular jobs, the runner will establish a http connection to the gitlab server. For the session server, Gitlab will establish a (tls-encrypted) connection to the runner, so you will need to expose the session service to Gitlab.

**Comment by kayman-mk** [comment]
Oh yes, you are absolutely right. I will add options for an IP address and a load balancer.

**Comment by roock** [comment]
The module already supports adding an EIP directly to the instance. Not sure if it is strictly required to put it behind a ALB, although it might be a more elegant solution.

**Comment by kayman-mk** [comment]
@roock Yeah, just saw it. So nothing to do for the EIP. An ALB listener can now be passed to the module. But the test is still missing. The idea is to use either EIP or ALB. Depends on your preferences, company guidelines, ...

**Comment by kayman-mk** [comment]
Ok, let's give it another try. Stick with EIP for the base feature and add ALB support later.

**Comment by connorshea** [comment]
Was there any progress toward this? :)

**Comment by kayman-mk** [comment]
No progress at all. Any ideas?
